docker/compute/parflow/Dockerfile

```diff
@@ -0,0 +1,81 @@
+# Docker file for Parflow
+
+ARG BASE_IMAGE=ubuntu:20.04
+
+FROM ${BASE_IMAGE} as build
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive \
+    apt-get install -y --no-install-recommends \
+        git \
+        wget \
+        ca-certificates \
+        build-essential \
+        gfortran \
+        libopenblas-dev \
+        liblapack-dev \
+        libopenmpi-dev \
+        openmpi-bin \
+        tcl-dev \
+        tk-dev \
+        cmake && \
+    rm -rf /var/lib/apt/lists/*
+
+# ParFlow environment variables
+ENV HYPRE_DIR=/hypre
+ENV PARFLOW_DIR=/parflow
+
+# Install hypre
+RUN wget -q https://github.com/hypre-space/hypre/archive/v2.18.2.tar.gz && \
+    tar xf v2.18.2.tar.gz && \
+    cd hypre-2.18.2/src && \
+    ./configure --prefix=/hypre --with-MPI && \
+    make install && \
+    cd / && \
+    rm -fr hypre-2.18.2 v2.18.2.tar.gz
+
+# Install parflow
+RUN git clone -b master --single-branch https://github.com/parflow/parflow.git /parflow-tmp
+
+RUN mkdir -p /build && \
+    cd /build && \
+    cmake ../parflow-tmp \
+       -DPARFLOW_AMPS_LAYER=mpi1 \
+       -DPARFLOW_AMPS_SEQUENTIAL_IO=TRUE \
+       -DHYPRE_ROOT=/hypre \
+       -DPARFLOW_ENABLE_TIMING=TRUE \
+       -DPARFLOW_HAVE_CLM=TRUE \
+       -DCMAKE_INSTALL_PREFIX=/parflow && \
+    make install && \
+    cd / && \
+    rm -fr /parflow-tmp /build
+
+FROM ${BASE_IMAGE}
+LABEL maintainer="patrick.avery@kitware.com" \
+      version="1.0" \
+      parflow.version="master"
+
+COPY --from=build /parflow /parflow
+
+ENV PARFLOW_DIR=/parflow
+
+# Install required runtime files
+RUN apt-get update && \
+    DEBIAN_FRONTEND=noninteractive \
+    apt-get install -y --no-install-recommends \
+        openmpi-bin \
+        libgfortran-9-dev \
+        openssh-server \
+        tcl && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV PARFLOW_MPIEXEC_EXTRA_FLAGS "--mca mpi_yield_when_idle 1 --oversubscribe"
+
+# Run as a non-root user
+RUN useradd -m -U -s /bin/bash user
```

Comment:
do we want the user to be `user` or `parflow`?

Reply:
It can be anything. Just let me know which one.

Reply:
I agree with anything, but I was just wondering if there was a convention of some kind somewhere? Any previous experience on that?

Should we use a uid of 1000 too?

Reply:
I am not sure about a convention for names. I haven't personally seen a common convention.

I think making a uid of 1000 is probably a good idea, so we get a consistent UID. It is also what [jupyter does](https://github.com/jupyter/docker-stacks/blob/76402a27fd13e649a3166b1da5a162ac476912bf/base-notebook/Dockerfile#L13-L14), and I think lots of people use their docker images. I guess it might be a good convention for us to use a UID of 1000 and a GID of 100.

Note that they also use "jovyan" for their user names, which is just what jupyter uses to describe [members of the jupyter community](https://discourse.jupyter.org/t/what-is-with-the-weird-jovyan-user/1673/3).